File: Heat/ViewModels/ThermalUnits/CreateThermalUnitViewModel.vb
Imports System.ComponentModel.DataAnnotations

Namespace ViewModels.ThermalUnits
    Public Class CreateThermalUnitViewModel

        Property PlantID As Integer
        <Display(name:="Impianto")> _
        Property PlantDescription As String

        <Display(name:="Marca")> _
        Property ManifacturerID As String
        Property ManifacturerList As IEnumerable(Of SelectListItem)

        <Display(name:="Modello")> _
        Property ModelID As Integer
        Property ModelList As IEnumerable(Of SelectListItem)

        <Display(name:="Matricola")> _
        Property SerialNumber As String

        <Display(name:="Data di installazione")> _
        <DataType(DataType.Date)> _
        Property InstallationDate As DateTime

        <Display(name:="Data di prima accensione")> _
        <DataType(DataType.Date)> _
        Property FirstStartUp As Nullable(Of DateTime)

        <Display(name:="Garanzia")> _
        Property Warranty As String

        <Display(name:="Scadenza garanzia")> _
        <DataType(DataType.Date)> _
        Property WarrantyExpiration As Nullable(Of DateTime)

        <Required> _
        <Display(name:="Combustibile")> _
        Property FuelID As Integer
        Property FuelList As IEnumerable(Of SelectListItem)

        <Display(name:="Potenza termica utile nominale Pn max (kW)")> _
        Property NominalThermalPowerMax As Single

        <Display(name:="Fluido termovettore")> _
        Property HeatTransferFluidID As Integer
        Property HeatTransferFluidList As IEnumerable(Of SelectListItem)

        <Display(name:="Rendimento termico utile a Pn max (%)")> _
        Property ThermalEfficiencyPowerMax As Single

        <Required> _
        <Display(name:="Tipo di unità")> _
        Property Kind As ThermalUnitKindEnum



    End Class
End Namespace

File: Heat.ConvertedToC#/ViewModels/ThermalUnits/CreateThermalUnitViewModel.cs
using System;
using System.Collections.Generic;
using System.Web.Mvc;
using System.ComponentModel.DataAnnotations;

namespace Heat.ViewModels.ThermalUnits
{
    public class CreateThermalUnitViewModel
	{

		public int PlantID { get; set; }
		[Display(Name = "Impianto")]
		public string PlantDescription { get; set; }
		public IEnumerable<SelectListItem> PlantList { get; set; }

		public bool PlantIDSelected { get; set; }

		[Display(Name = "Marca")]
		public string ManifacturerID { get; set; }
		public IEnumerable<SelectListItem> ManifacturerList { get; set; }

		[Display(Name = "Modello")]
		public int ModelID { get; set; }
		public IEnumerable<SelectListItem> ModelList { get; set; }

		[Display(Name = "Matricola")]
		public string SerialNumber { get; set; }

		[Display(Name = "Data di installazione")]
		[DataType(DataType.Date)]
		public DateTime InstallationDate { get; set; }

		[Display(Name = "Data di prima accensione")]
		[DataType(DataType.Date)]
		public Nullable<DateTime> FirstStartUp { get; set; }

		[Display(Name = "Garanzia")]
		public string Warranty { get; set; }

		[Display(Name = "Scadenza garanzia")]
		[DataType(DataType.Date)]
		public Nullable<DateTime> WarrantyExpiration { get; set; }

		[Required()]
		[Display(Name = "Combustibile")]
		public int FuelID { get; set; }
		public IEnumerable<SelectListItem> FuelList { get; set; }

		[Display(Name = "Potenza termica utile nominale Pn max (kW)")]
		public float NominalThermalPowerMax { get; set; }

		[Display(Name = "Fluido termovettore")]
		public int HeatTransferFluidID { get; set; }
		public IEnumerable<SelectListItem> HeatTransferFluidList { get; set; }

		[Display(Name = "Rendimento termico utile a Pn max (%)")]
		public float ThermalEfficiencyPowerMax { get; set; }

		[Required()]
		[Display(Name = "Tipo di unità")]
		public ThermalUnitKindEnum Kind { get; set; }



	}
}

